optimizely/config/polling_manager.go

```diff
@@ -53,42 +53,42 @@ type PollingProjectConfigManager struct {
 	exeCtx utils.ExecutionCtx // context used for execution control
 }
 
-func (cm *PollingProjectConfigManager) activate(initialPayload []byte, init bool) {
-
-	update := func() {
-		var e error
-		var code int
-		var payload []byte
-		if init && len(initialPayload) > 0 {
-			payload = initialPayload
-		} else {
-			payload, code, e = cm.requester.Get()
-
-			if e != nil {
-				cmLogger.Error(fmt.Sprintf("request returned with http code=%d", code), e)
-			}
-		}
-
-		projectConfig, err := datafileprojectconfig.NewDatafileProjectConfig(payload)
-		if err != nil {
-			cmLogger.Error("failed to create project config", err)
+// SyncConfig gets current datafile and updates projectConfig
+func (cm *PollingProjectConfigManager) SyncConfig(datafile []byte) {
+	var e error
+	var code int
+	if len(datafile) == 0 {
+		datafile, code, e = cm.requester.Get()
+
+		if e != nil {
+			cmLogger.Error(fmt.Sprintf("request returned with http code=%d", code), e)
 		}
+	}
 
-		cm.configLock.Lock()
-		cm.projectConfig = projectConfig
-		cm.err = err
-		cm.configLock.Unlock()
+	// TODO: Compare revision numbers here and set projectConfig only if the revision number has changed
+	projectConfig, err := datafileprojectconfig.NewDatafileProjectConfig(datafile)
+	if err != nil {
+		cmLogger.Error("failed to create project config", err)
 	}
 
+	cm.configLock.Lock()
+	cm.projectConfig = projectConfig
+	cm.err = err
+	cm.configLock.Unlock()
+}
+
+func (cm *PollingProjectConfigManager) start(initialDatafile []byte, init bool) {
+
 	if init {
-		update()
+		cm.SyncConfig(initialDatafile)
 		return
 	}
+
 	t := time.NewTicker(cm.pollingInterval)
 	for {
 		select {
 		case <-t.C:
-			update()
+			cm.SyncConfig([]byte{})
```

Comment:
Pass in nil instead

Reply:
For now it can be fine (passing bytes or nil), but I think CM should eventually have this info.
In my opinion, both sdk key and datafile url have to be the properties of CM. so SyncConfig will need to change to something like PollAndSync, and we can have a similar method (StaticSync) with provided bytes.

Reply:
I am going by @pawels-optimizely 's comment and leaving this as is.